Classes/URKArchive.mm

```diff
@@ -385,23 +395,75 @@ + (BOOL)urlIsARAR:(NSURL *)fileURL
     return [NSArray arrayWithArray:fileInfos];
 }
 
-- (nullable NSString *)firstVolumePath {
-    return @"";
+- (NSString *)firstVolumePath:(NSString *)filePath {
+    NSURL *fileURL = [self firstVolumeURL:[NSURL fileURLWithPath:filePath]];
+
+    return fileURL.path;
 }
 
-- (nullable NSURL *)firstVolumeURL {
-    NSString *path = [self firstVolumePath];
+- (NSURL *)firstVolumeURL:(NSURL *)fileURL {
     
-    if (![path length]) {
-        return nil;
+    URKLogDebug("Checking if the archive is part of a volume...");
+    
+    __block NSString *volumePath = fileURL.path;
+    NSTextCheckingResult * match;
+    
+    if (volumePath.length)
```

Comment:
Invert this `if` and exit early. If `volumePath` is empty, log an error and return nil. That way we un-nest everything below, and you don't need to check for `match` twice. You can simplify the rest of the method's conditions.